src/textual/widgets/_tree.py

```diff
@@ -515,6 +515,26 @@ def get_node_at_line(self, line_no: int) -> TreeNode[TreeDataType] | None:
         else:
             return line.node
 
+    class UnknownID(Exception):
```

Comment:
Contrary to what I said before, I'm thinking that `UnknownNodeID` would be better. We have other IDs in the code that may confuse.

Reply:
Sounds good to me.

Reply:
Done.